python/mxnet/metric.py

```diff
@@ -65,20 +106,27 @@ def get(self):
 class Accuracy(EvalMetric):
     """Calculate accuracy"""
 
-    def __init__(self):
+    def __init__(self, **kwargs):
         super(Accuracy, self).__init__('accuracy')
+        try:
+            self.top_k = kwargs['top_k']
+        except KeyError:
+            self.top_k = 1
+        self.name += '-Top-%d' % self.top_k
 
     def update(self, labels, preds):
         check_label_shapes(labels, preds)
 
         for i in range(len(labels)):
-            pred_label = ndarray.argmax_channel(preds[i]).asnumpy().astype('int32')
+            pred_label = numpy.argsort(preds[i].asnumpy().astype('float32'), axis = 1)
```

Comment:
ndarray.argmax_channel allows computing prediction on GPU, this is critical for performance when output is large, e.g. for segmentation.
Please keep it unless top_k > 1.